app/pods/agenda/route.js

```diff
@@ -25,10 +25,15 @@ export default class AgendaRoute extends Route {
       sort: '-serialnumber',
       include: 'status',
     });
+    const agendaitems = await this.store.query('agendaitem', {
+      'filter[agenda][:id:]': agenda.id,
+      sort: 'show-as-remark,number',
+    });
```

Comment:
Currently, the only reason for this is the getter `firstAgendaitem` in `agenda/agendaitems`.
The route `agenda/agendaitems` has its own query for agendaitems and is not the same.

https://github.com/kanselarij-vlaanderen/frontend-kaleidos/blob/feature/KAS-2667-new-agenda-overview/app/pods/agenda/agendaitems/route.js#L33

Loading agendaitems again here will slow down the overall process of loading the agenda, impacting performance.
This query is not cached by `cache-warmup-service`, unlike the one in the other route.
Also I wouldn't recommend loading agendaitems in the top route, since you can visit `/documents` or `/compare` routes and you don't need the agendaitems loaded in that case. It makes more sense to keep them in the route `agenda/agendaitems`.

Either queryOne agendaitem as the `firstAgendaitem` and pass that down, or just load it with a task where you need it in `AgendaTabs` since that is the only place that needs it.

Reply:
I load the agendaitems here in the agenda route because we do need them, for rendering the `agenda-tabs` in the header. These tabs are rendered anytime an agenda is opened, even in the `/documents` and `/compare` subroutes. The reason for passing down the agendaitems to the component is to force it ro re-render when the agendaitems change. E.g. when you delete the last agendaitem in an agenda, the "Details" tab still exists because the task to load the agendaitems does not refire. Then clicking on that tab breaks the app. See: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1288/files/e80d2828b48d5cdd442a1d937d23051383f008bd#r851374926

Didn't realize it got loaded in again down the line, but the query in this route can be made conform with the `agenda.agendaitems` query, and in the `agenda.agendaitems` route we get it from the `agenda` model.